test/smartwallet/nativeHolderSmartWallet.test.ts

```diff
@@ -1,401 +1,398 @@
+      });
+
+      const execution = async () => {
+        await expect(
+          smartWallet
+            .connect(relayHub)
+            .execute(
+              suffixData,
+              relayRequest.request,
+              worker.address,
+              signature
+            )
+        ).not.to.be.rejected;
+
+        expect(fakeToken.transfer, 'Token.transfer() was not called').to.be
+          .called;
+        expect(recipientContract.emitMessage, 'Recipient method was not called')
+          .to.be.called;
+      };
+
+      await expectBalanceAfterSuccessExecution(
+        recipientContract.address,
+        TWO_ETHERS,
+        execution
+      );
+    });
+
+    it('Should fail if the smart wallet cannot pay native currency', async function () {
+      // we try to transfer a value higher than the SW balance
+      const valueToTransfer = INITIAL_SW_BALANCE.add(
+        ethers.utils.parseEther('1')
+      ).toString();
+      const { relayRequest, suffixData, signature } = prepareRequest({
+        data: encodedFunctionData,
+        value: valueToTransfer.toString(),
+      });
+
+      const execution = async function () {
+        // Use the static call to check the returned values
+        const tx = await smartWallet
+          .connect(relayHub)
+          .callStatic.execute(
+            suffixData,
+            relayRequest.request,
+            worker.address,
+            signature
+          );
+        expect(tx.success, 'Success is true').to.be.false;
+
+        await expect(
+          smartWallet
+            .connect(relayHub)
+            .execute(
+              suffixData,
+              relayRequest.request,
+              worker.address,
+              signature
+            ),
+          'Execute transaction rejected'
+        ).not.to.be.rejected;
+      };
+
+      const recipientBalancePriorExecution = await provider.getBalance(
+        recipientContract.address
+      );
+      const swBalancePriorExecution = await provider.getBalance(
+        smartWallet.address
+      );
+
+      await expectBalanceToBeRight(
+        execution,
+        recipientContract.address,
+        recipientBalancePriorExecution,
+        swBalancePriorExecution
+      );
+    });
+  });
+
+  describe('Function directExecute(address to,uint256 value,bytes calldata data)', function () {
+    it('Should transfer native currency without executing transactions', async function () {
+      const execution = async () => {
+        await expect(
+          smartWallet['directExecute(address,uint256,bytes)'](
+            recipientContract.address,
+            TWO_ETHERS,
+            '0x00'
+          ),
+          'Execution failed'
+        ).not.to.be.rejected;
+      };
+      await expectBalanceAfterSuccessExecution(
+        recipientContract.address,
+        TWO_ETHERS,
+        execution
+      );
+    });
+
+    it('Should transfer native currency and execute transactions', async function () {
```

Comment:
Based on test description, I think it should assert that the emmitMessage has been called.

Reply:
Good catch, thank you!